File: legion_test/legion_test/robot/feedback.py
#
#    Copyright 2018 EPAM Systems
#
#    Licensed under the Apache License, Version 2.0 (the "License");
#    you may not use this file except in compliance with the License.
#    You may obtain a copy of the License at
#
#        http://www.apache.org/licenses/LICENSE-2.0
#
#    Unless required by applicable law or agreed to in writing, software
#    distributed under the License is distributed on an "AS IS" BASIS,
#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#    See the License for the specific language governing permissions and
#    limitations under the License.
#
"""
Robot test library - feedback
"""
import json

import requests


class Feedback:
    """
    Feedback client for robot tests
    """

    @staticmethod
    def send_feedback_data(domain, event, time=None, **payload):
        """
        Connect to grafana server

        :param domain: feedback domain
        :type domain: str
        :param event: event name
        :type event: str
        :param time: timestamp, by default - skipped
        :type time: str
        :param payload: payload data
        :type payload: str
        :return: None
        """
        url = '{}/{}'.format(domain, event)

        data = {
            'json': json.dumps(payload if payload else {})
        }

        if time and time != 'now':
            data['time'] = time

        response = requests.get(url, data=data)

        if response.status_code != 200:
            raise requests.RequestException('HTTP Code %d for "GET %s "' % (response.status_code, url))

File: legion/operator/pkg/webserver/routes/v1/feedback/feedback.go
/*
 * Copyright 2019 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package feedback

// Currently we can not aggregate swagger documentation from multiply services
// TODO: need to implement it

type FeedbackRequest struct{}
type FeedbackResponse struct{}

// @Summary Send feedback about previously made prediction
// @Description Send feedback about previously made prediction
// @Tags Feedback
// @Param feedback body feedback.FeedbackRequest true "Feedback Request"
// @Accept  json
// @Produce  json
// @Param model-name header string true "Model name"
// @Param model-version header string true "Model version"
// @Param request-id header string true "Request ID"
// @Success 200 {object} feedback.FeedbackResponse
// @Router /api/v1/feedback [post]
func stubFeedback() {
	panic("must be never invoked!")
}
